Kinvey.Core/Query/KinveyQueryVisitor.cs

```diff
@@ -112,31 +112,65 @@ public override void VisitWhereClause(WhereClause whereClause, QueryModel queryM
 			{
 				BinaryExpression equality = whereClause.Predicate as BinaryExpression;
 				var member = equality.Left as MemberExpression;
+				var argument = equality.Right.ToString();
 
-				builderMongoQuery.Write ("\"" + mapPropertyToName[member.Member.Name] + "\"");
-				builderMongoQuery.Write (":");
-				builderMongoQuery.Write (equality.Right);
+				builderMongoQuery.Write("\"" + mapPropertyToName[member.Member.Name] + "\"");
+				builderMongoQuery.Write(":");
+
+				if (equality.Right.Type.Name.Equals("Boolean"))
```

Comment:
@tejasranade In general I agree, but in this case, an expression tree is being traversed, and if there was an issue with forming that tree, it would be caught at compile time. At runtime, this should be a properly formed expression.